File: Python/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices):
        if len(prices) < 1:
            return 0
        min_price = prices[0]
        max_profit = 0
        for price in prices:
            if price - min_price > max_profit:
                max_profit = price - min_price
            if price < min_price:
                min_price = price

        return max_profit

File: C/best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize){
    int i;
    int now, best;

    if (pricesSize == 0) return 0;

    for (i = 1, now = prices[0], best = 0; i < pricesSize; ++i) {
        if (prices[i] - now > best) best = prices[i] - now;
        if (prices[i] < now) now = prices[i];
    }
    return best;
}
